### DISCUSSION TITLE
Provide Lazy Loading Component

### DISCUSSION THREAD

**Discussion by qingant** [zauberzeug/nicegui#1452] [Ideas / Feature Requests]
It sounds like the slowness of NiceGUI on Safari is just because too many things to render. Should we provide a lazy loading component that render things only when it's about to be scroll to visible area ?

**Comment by falkoschindler** [comment]
Interesting idea, @qingant!
Reminds me of the [intersection_observer](https://github.com/zauberzeug/nicegui/blob/main/website/intersection_observer.py) we use on our website to run the demos lazily.

**Reply by qingant** [reply]
thanks @falkoschindler  
it seems very difficult to implement a general lazy loading component without know the size of component prior to rendering.

**Reply by falkoschindler** [reply]
@qingant I'm not sure. It's certainly hard to tell _exactly_ if an element intersects with the viewport. But you can start rendering elements when they are _near_ the viewport, e.g. one screen height below the bottom edge. This makes their height irrelevant.
